PubNub/Data/Service Objects/PNResult.m

```diff
@@ -83,22 +83,29 @@ - (id)copyWithZone:(NSZone *)zone {
     result.authKey = self.authKey;
     result.origin = self.origin;
     result.clientRequest = self.clientRequest;
-    result.serviceData = self.serviceData;
-    
+    [result updateData:self.serviceData];
+
     return result;
 }
 
 - (instancetype)copyWithMutatedData:(id)data {
     
     PNResult *result = [self copy];
-    result->_serviceData = [data copy];
-    
+    [result updateData:data];
+
     return result;
 }
 
++ (NSDictionary *)normalizeServiceData:(id)serviceData {
+    if (serviceData && ![serviceData isKindOfClass:[NSDictionary class]]) {
+        return @{@"information": [serviceData description]};
```

Comment:
I think we should still do if([serviceData respondsToSelectod:@selector(copy)]) and then add the copy and not the description.